### PR TITLE
Move test program to submodule

### PR DESCRIPTION
**Pull request by slhmy**
Solve #85.

Additionally fixed `SETUP_ENV` bash.
Dev env should be usable for `cargo test` if you run `git submodule update --init --recursive`.

### CODE REVIEW

**Comment by KanadeSiina** [comment]
It seems good, but I think some guides should be added to `README.md` so that it could help others build the developing environment from scratch. 
BTW, though it is enough now, but I hope that we can remove the binary files as well as the `testlib.h` in the future.

File: .github/workflows/rust_build.yml [hunk 1]
```diff
@@ -2,27 +2,28 @@ name: Build
...
-      run: sudo apt-get install libseccomp-dev protobuf-compiler
-    - name: Test Command sh
-      run: sh -c "echo test"
-    - name: Check ENV
-      run: echo $(rustup --version && gcc -v)
-    - name: Build
-      run: cargo build --verbose
-    - name: Run tests
-      run: cargo test --verbose
+      - uses: actions/checkout@v2
```
Comment (cubercsl) [inline .github/workflows/rust_build.yml:17]:
```suggestion
      - uses: actions/checkout@v3
        with:
          submodules: recursive
```

File: .github/workflows/rust_build.yml [hunk 2]
```diff
@@ -2,27 +2,28 @@ name: Build
...
+      - name: Install dependencies
+        run: sudo apt-get install -y libseccomp-dev protobuf-compiler
+      - name: Prepare submodule
+        run: git submodule update --init --recursive
```
Comment (cubercsl) [inline .github/workflows/rust_build.yml:21]:
```suggestion
```
submodules should be checked out by `actions/checkout`

**Review by cubercsl — CHANGES_REQUESTED** [review]

File: .github/workflows/rust_build.yml (hunk 1 above)
Comment (slhmy) [reply .github/workflows/rust_build.yml]: Thanks for suggestion, changed.

**Comment by slhmy** [comment]
I'm merging this PR right now, for the known issues can be fixed later.

**Pull request merged**